Compiler/src/ssair/inlining.jl

```diff
@@ -357,8 +367,31 @@ function adjust_boundscheck!(inline_compact::IncrementalCompact, idx′::Int, st
     return nothing
 end
 
+function apply_gc_preserve_inlining!(insert_node!::Inserter, inline_compact::IncrementalCompact, line, @nospecialize(stmt′), @nospecialize(gc_preserve))
+    if isexpr(stmt′,  :invoke_modify)
+        # TODO: The combination of an existing invoke_modify and gc_preserve causes trouble. Bail for now.
```

Comment:
This is why I wasn't particularly in favor of adding a different, unrelated meaning to `invoke_modify`, instead of just adding some other metadata representation. I am somewhat curious if we can make the initial `GCPreserveDuring` implementation here go fully to being a `CallOperand` builtin function wrapper.

Reply:
You mean that the codegen for `invoke` would recognize it when doing a specfun call and automatically strip it? I think that's a reasonable resolution to the TODO also, but I was a little worried of what would happen if that type information got lost shomehow.

Reply:
That proposal actually versions the generated code though, where here we want to version the caller, but specifically not the generated code, so while there are some similarities, I think the mechanisms are complements of each other.